infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/util/EntryPointParser.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.environment.util;
+
+import static com.fasterxml.jackson.databind.node.JsonNodeType.STRING;
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.eclipse.che.api.core.model.workspace.config.MachineConfig;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.commons.annotation.Nullable;
+
+/**
+ * Can be used to parse container entry-point definition specified as a YAML list of strings or a
+ * single YAML string.
+ */
+public final class EntryPointParser {
+  private final YAMLMapper mapper = new YAMLMapper();
+
+  /**
+   * Parses the attributes contained in the provided machine config and produces an entry point
+   * definition.
+   *
+   * @param machineAttributes the attributes of a machine to extract the entry point info from
+   * @return an entry point definition, never null
+   * @throws InfrastructureException on failure to parse the command or arguments
+   */
+  public EntryPoint parse(Map<String, String> machineAttributes) throws InfrastructureException {
+    String command = machineAttributes.get(MachineConfig.CONTAINER_COMMAND_ATTRIBUTE);
+    String args = machineAttributes.get(MachineConfig.CONTAINER_ARGS_ATTRIBUTE);
+
+    List<String> commandList =
+        command == null
+            ? emptyList()
+            : parseAsList(command, MachineConfig.CONTAINER_COMMAND_ATTRIBUTE);
+    List<String> argList =
+        args == null ? emptyList() : parseAsList(args, MachineConfig.CONTAINER_ARGS_ATTRIBUTE);
+
+    return new EntryPoint(commandList, argList);
+  }
+
+  /**
+   * Parses a single entry in the format expected by the entry point attribute, i.e. a YAML list of
+   * strings.
+   *
+   * <p>The {@code null} return value is meant to be reinterpreted as some kind of exception with a
+   * more useful error message than the one we could provide here.
+   *
+   * @param data the string data to parse
+   * @return a parsed list or null if the data is not a YAML list
+   * @throws IllegalArgumentException if the parsing of the {@code entry} failed for other reasons
+   */
+  @VisibleForTesting
+  @Nullable
+  List<String> parseEntry(String data) throws IllegalArgumentException {
```

Comment:
Do we still need this method?
Can not we rewrite `parseAsList` in the following way:
```java
  private List<String> parseAsList(String data, String attributeName)
      throws InfrastructureException {
    try {
      return mapper.readValue(data, new TypeReference<ArrayList<String>>() {});
    } catch (IllegalArgumentException | IOException e) {
      throw new InfrastructureException(
          format(
              "Failed to parse the attribute %s as a YAML list. The value was: %s. Error: %s",
              attributeName, data, e.getMessage()), e);
    }
  }
```